cmake/modules/FindTexturePacker.cmake

```diff
@@ -9,44 +9,78 @@
 #
 # This will define the following (imported) targets::
 #
-#   TexturePacker::TexturePacker   - The TexturePacker executable
+#   TexturePacker::TexturePacker::Executable   - The TexturePacker executable participating in build
+#   TexturePacker::TexturePacker::Installable  - The TexturePacker executable shipped in the Kodi package
 
-if(NOT TARGET TexturePacker::TexturePacker)
+if(NOT TARGET TexturePacker::TexturePacker::Executable)
   if(KODI_DEPENDSBUILD)
     get_filename_component(_tppath "${NATIVEPREFIX}/bin" ABSOLUTE)
     find_program(TEXTUREPACKER_EXECUTABLE NAMES "${APP_NAME_LC}-TexturePacker" TexturePacker
                                           HINTS ${_tppath})
 
-    add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
-    set_target_properties(TexturePacker::TexturePacker PROPERTIES
-                                                       IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
+    add_executable(TexturePacker::TexturePacker::Executable IMPORTED GLOBAL)
+    set_target_properties(TexturePacker::TexturePacker::Executable PROPERTIES
+                                          IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
   elseif(WIN32)
     get_filename_component(_tppath "${DEPENDENCIES_DIR}/tools/TexturePacker" ABSOLUTE)
     find_program(TEXTUREPACKER_EXECUTABLE NAMES "${APP_NAME_LC}-TexturePacker.exe" TexturePacker.exe
                                           HINTS ${_tppath})
 
-    add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
-    set_target_properties(TexturePacker::TexturePacker PROPERTIES
-                                                       IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
+    add_executable(TexturePacker::TexturePacker::Executable IMPORTED GLOBAL)
+    set_target_properties(TexturePacker::TexturePacker::Executable PROPERTIES
+                                          IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
   else()
+    set(TEXTUREPACKER_EXECUTABLE "TEXTUREPACKER_EXECUTABLE-NOTFOUND")
```

Comment:
this shouldn't be needed, it's set by find_program

Reply:
If `WITH_TEXTUREPACKER` is not defined, `find_program` is not invoked and `TEXTUREPACKER_EXECUTABLE` is empty. If `WITH_TEXTUREPACKER` is defined, `find_program` returns either the path to TexturePacker or `TEXTUREPACKER_EXECUTABLE-NOTFOUND`, but not an empty string! And `if(TEXTUREPACKER_EXECUTABLE)` will always evaluate to true because `TEXTUREPACKER_EXECUTABLE` is not empty.

Reply:
If the var is TEXTUREPACKER_EXECUTABLE-NOTFOUND, if evaluates to false.
Not very intuitive, but that's how cmake works